### PR TITLE
Merge main into ray/ui-update

### PR DESCRIPTION
**Pull request by rjrjr**
This was mostly a mechanical merge except for the poetry benchmark app. @steve-the-edwards, can you confirm I didn't break that? Also curious about your and @RBusarow's reaction to the conversion of `MayBeLoadingScreen` to a `typealias`, and our ability to delete the `LoadingContainer` View subclass entirely.

* origin/main:
  Bump tutorials to 1.7.0
  Bump okio to 3.0.0
  Finish releasing v1.7.0
  Releasing v1.7.0
  Fixes a typo in step 1 of RELEASING.md
  Exclude Benchmarks for connectedCheck verification instructions
  Fixes memory leak in ViewStateCache.
  Update AGP versions and OptIn Notation
  Refactor Perf Poetry to add Related Benchmarks
  699: Add simulated performance config to Poetry
  Allow repeatOnLifecycle state to be configurable.

### CODE REVIEW

File: benchmarks/performance-poetry/complex-poetry/src/main/java/com/squareup/benchmarks/performance/complex/poetry/views/LoaderSpinner.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import com.squareup.workflow1.ui.AndroidScreen
+import com.squareup.workflow1.ui.ScreenViewFactory
+import com.squareup.workflow1.ui.ScreenViewHolder
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.container.OverlayDialogFactory
+import kotlin.reflect.KClass
+
+@OptIn(WorkflowUiExperimentalApi::class)
+object LoaderSpinner : AndroidOverlay<LoaderSpinner> {
```
Comment (rjrjr) [inline benchmarks/performance-poetry/complex-poetry/src/main/java/com/squareup/benchmarks/performance/complex/poetry/views/LoaderSpinner.kt:18]: @steve-the-edwards This replaces `LoaderContainer`. I'm not thrilled that it requires the author to instantiate the Dialog themselves, where with the old one they could just focus on building a view as usual. 

I'm inclined to follow up by adding a vanilla `Overlay` wrapper type that makes any arbitrary screen a full bleed dialog, but that's too much for a merge PR.

**Comment by rjrjr** [comment]
> This was mostly a mechanical merge except for the poetry benchmark app. @steve-the-edwards, can you confirm I didn't break that?

Oops, I did, crashes on navigation. Following up…

**Review by steve-the-edwards — APPROVED** [review]
Nice. All makes sense to me. I'll look out for it when you merge back as I have a PR in flight.

**Comment by rjrjr** [comment]
Not sure what's going on. `LoaderSpinner` implements `AndroidOverlay` but the `as` check to that effect in `OverlayDialogFactoryFinder` is failing.

**Comment by rjrjr** [comment]
Oops:
Screen Shot 2022-04-14 at 12 56 06 PM

**Comment by rjrjr** [comment]
Fixed benchmark poetry. It's spinner is badly sized, another reason to follow up with the Overlay work suggested above. Chatted with Steve, will merge this as is in the meantime.

**Pull request merged**